client/my-sites/stats/stats-download-csv/index.jsx

```diff
@@ -87,7 +86,7 @@ const connectComponent = connect( ( state, ownProps ) => {
 		isLoading = dataList.isLoading();
 	} else {
 		data = getSiteStatsCSVData( state, siteId, statType, query );
-		isLoading = isRequestingSiteStatsForQuery( state, siteId, statType, query );
+		isLoading = isRequestingSiteStatsForQuery( state, siteId, statType, query ) && ! data;
```

Comment:
I'm not sure the `isLoading` is working as expected to disable the button.

On a large site, I was able to visit the "Year" summary view & click the "Download Data as CSV" button before there was usable data in place (i.e. immediately & before the chart and viz had activated).

I was served an empty CSV file.

Well, mostly empty. The file contained a single BOM / Byte order mark / U+FEFF character. That same character is present in a data-populated CSV, so probably not related to this change.